### ISSUE TITLE
Not able to `make hungarian` during installation

### ISSUE THREAD

**Issue by abhijayghildyal** [Russell91/TensorBox#56]
`cd /path/to/tensorbox/utils && make && make hungarian && cd ..`

Error:
make: *** No rule to make target `'hungarian/hungarian.so'`, needed by `'hungarian'`.  Stop.

and when I cd into /path/to/tensorbox/utils/hungarian and run `make hungarian`, it gives me the following error:

      g++     hungarian.cc   -o hungarian
      hungarian.cc:2:42: fatal error: tensorflow/core/framework/op.h: No such file or directory
      #include "tensorflow/core/framework/op.h"
                                                ^
      compilation terminated.
      make: *** [hungarian] Error 1

I am trying to install it in two different machines with ubuntu 14.04, 64 bit, python 2.7
1. cpu-only `tensorflow` version is `0.10.0rc0`
2. GPU enabled `tensorflow` version is `0.9.0`

**Comment by Russell91** [comment]
Have you cloned the latest repository? The readme was changed just
yesterday.

**Comment by abhijayghildyal** [comment]
I cloned again and followed the instructions. It works now. Thanks

**Comment by abhijayghildyal** [comment]
I cloned again from the repository and it works now.

**Comment by abhijayghildyal** [comment]
I cloned the repository again and this time my `tensorflow` version is `0.8.0rc0` ( CPU - only )

`cd ~/russel91_Tensorbox/tensorbox/utils && make && make hungarian && cd ..`

Similar error persists 

    pip install runcython
    Requirement already satisfied (use --upgrade to upgrade): runcython in /usr/local/lib/python2.7/dist-   packages
    Requirement already satisfied (use --upgrade to upgrade): Cython>=0.10 in     /usr/local/lib/python2.7/dist-packages (from runcython)
    Cleaning up...
    makecython++ stitch_wrapper.pyx "" "stitch_rects.cpp ./hungarian/hungarian.cpp"
    make: *** No rule to make target \`hungarian/hungarian.so', needed by `hungarian'.  Stop.

**Comment by abhijayghildyal** [comment]
This is the error it gives me when i run the training without running `make hungarian`

    python train.py --hypes hypes/lstm_rezoom.json --gpu 0 --logdir output

I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcublas.so locally
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcudnn.so locally
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcufft.so locally
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcuda.so locally
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcurand.so locally
I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:924] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
I tensorflow/core/common_runtime/gpu/gpu_init.cc:102] Found device 0 with properties: 
name: GeForce GT 640
major: 3 minor: 0 memoryClockRate (GHz) 0.9015
pciBusID 0000:05:00.0
Total memory: 2.00GiB
Free memory: 1.98GiB
I tensorflow/core/common_runtime/gpu/gpu_init.cc:126] DMA: 0 
I tensorflow/core/common_runtime/gpu/gpu_init.cc:136] 0:   Y 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:806] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GT 640, pci bus id: 0000:05:00.0)
WARNING:tensorflow:<tensorflow.python.ops.rnn_cell.BasicLSTMCell object at 0x7fdfa519e750>: Using a concatenated state is slower and will soon be deprecated.  Use state_is_tuple=True.
Traceback (most recent call last):
  File "train.py", line 533, in <module>
    main()
  File "train.py", line 530, in main
    train(H, test_images=[])
  File "train.py", line 443, in train
    smooth_op, global_step, learning_rate, encoder_net) = build(H, q)
  File "train.py", line 336, in build
    boxes_loss[phase]) = build_forward_backward(H, x, encoder_net, phase, boxes, flags)
  File "train.py", line 242, in build_forward_backward
    hungarian_module = tf.load_op_library('utils/hungarian/hungarian.so')
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/load_library.py", line 71, in load_op_library
    raise errors._make_specific_exception(None, None, error_msg, error_code)
tensorflow.python.framework.errors.NotFoundError: utils/hungarian/hungarian.so: cannot open shared object file: No such file or directory

**Comment by abhijayghildyal** [comment]
I tried running 

    g++ -std=c++11 -shared hungarian.cc -o hungarian.so -fPIC -I /usr/local/lib/python2.7/dist-packages/tensorflow/include

instead of

    TF_INC=$$(python -c 'import tensorflow as tf; print(tf.sysconfig.get_include())') && \
	g++ -std=c++11 -shared hungarian.cc -o hungarian.so -fPIC -I $$TF_INC
and it did not make the output file `hungarian.so`

**Comment by Russell91** [comment]
What was the error message when you typed that in manually...

**Comment by abhijayghildyal** [comment]
There was no error message.

**Comment by Russell91** [comment]
Can you explain

**Comment by nitish11** [comment]
@Russell91 : Even I am getting similar error with the new build. Please check the snippet below from my Ubuntu 14.04 terminal.

```
$ make
pip install runcython
Requirement already satisfied (use --upgrade to upgrade): runcython in /usr/local/lib/python2.7/dist-packages
Requirement already satisfied (use --upgrade to upgrade): Cython>=0.10 in /usr/local/lib/python2.7/dist-packages (from runcython)
makecython++ stitch_wrapper.pyx "" "stitch_rects.cpp ./hungarian/hungarian.cpp"
$ make hungarian
make: *** No rule to make target `hungarian/hungarian.so', needed by `hungarian'.  Stop.
```

**Comment by abhijayghildyal** [comment]
I installed tensorflow using pip, version 0.9 gpu enabled

After that I cloned the repository and followed the installation instructions. OverFeat algorithm ran properly. For reinspect installations when I ran the command `make hungarian` it gave me the error as mentioned by @nitish11. So I tried to manually run the following given in `Makefile`:

    TF_INC=$$(python -c 'import tensorflow as tf; print(tf.sysconfig.get_include())') && \
        g++ -std=c++11 -shared hungarian.cc -o hungarian.so -fPIC -I $$TF_INC

as

    g++ -std=c++11 -shared hungarian.cc -o hungarian.so -fPIC -I /usr/local/lib/python2.7/dist-packages/tensorflow/include

which ran without errors but no output which is the `hungarian.so` file.

**Comment by Russell91** [comment]
@nitish11 Okay, thank you for making this clear. The makefile rule failed when hungarian.so did not already exist. Can you try the latest makefile from here?

https://github.com/Russell91/TensorBox/commit/170c2fbd2c9c42998a51cfccbe7ea06a7f46b122

**Comment by nitish11** [comment]
@Russell91 :+1:  thanks. It's working now.

@abhijayghildyal : Expected output will be as follows.

```
$ make && make hungarian
pip install runcython
Requirement already satisfied (use --upgrade to upgrade): runcython in /usr/local/lib/python2.7/dist-packages
Requirement already satisfied (use --upgrade to upgrade): Cython>=0.10 in /home/nitish/.local/lib/python2.7/site-packages (from runcython)
Cleaning up...
makecython++ stitch_wrapper.pyx "" "stitch_rects.cpp ./hungarian/hungarian.cpp"
cd hungarian && \
	TF_INC=$(python -c 'import tensorflow as tf; print(tf.sysconfig.get_include())') && \
	g++ -std=c++11 -shared hungarian.cc -o hungarian.so -fPIC -I $TF_INC

```

**Comment by abhijayghildyal** [comment]
Thanks @Russell91 it's solved now.

**Comment by Zehaos** [comment]
Hi @Russell91，I got an error different from above, do you have some idea?
I use the latest Tensorflow (v0.11.0) and Tensorbox. Error message below.
```
zehao@lab:~/WorkSpace/TensorBox/utils$ make hungarian
cd hungarian && \
	TF_INC=$(python -c 'import tensorflow as tf; print(tf.sysconfig.get_include())') && \
	if [ `uname` == Darwin ];\
	then g++ -std=c++11 -shared hungarian.cc -o hungarian.so -fPIC -I -D_GLIBCXX_USE_CXX11_ABI=0$TF_INC;\
	else g++ -std=c++11 -shared hungarian.cc -o hungarian.so -fPIC -I  $TF_INC; fi
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcublas.so locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcudnn.so.5 locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcufft.so locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcuda.so.1 locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcurand.so locally
In file included from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h:101:0,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/strings/str_util.h:23,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:29,
                 from hungarian.cc:2:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:232:38: error: ‘tensorflow::gtl::array_slice_internal::ArraySliceImplBase<const T>::ArraySliceImplBase’ names constructor
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:252:32: error: ‘tensorflow::gtl::array_slice_internal::ArraySliceImplBase<T>::ArraySliceImplBase’ names constructor
In file included from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h:102:0,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/strings/str_util.h:23,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:29,
                 from hungarian.cc:2:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/inlined_vector.h: In member function ‘void tensorflow::gtl::InlinedVector<T, N>::Destroy(T*, int)’:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/inlined_vector.h:394:10: error: ‘is_trivially_destructible’ is not a member of ‘std’
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/inlined_vector.h:394:42: error: expected primary-expression before ‘>’ token
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/inlined_vector.h:394:43: error: ‘::value’ has not been declared
In file included from hungarian.cc:3:0:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/op_kernel.h: At global scope:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/op_kernel.h:169:19: error: ‘tensorflow::OpKernel::OpKernel’ names constructor
In file included from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/strings/str_util.h:23:0,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:29,
                 from hungarian.cc:2:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h: In instantiation of ‘tensorflow::gtl::ArraySlice<T>::ArraySlice(const tensorflow::gtl::InlinedVector<T, N>&) [with int N = 4; T = tensorflow::DataType]’:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/types.h:86:36:   required from here
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h:140:33: error: no matching function for call to ‘tensorflow::gtl::array_slice_internal::ArraySliceImpl<tensorflow::DataType>::ArraySliceImpl(tensorflow::gtl::InlinedVector<tensorflow::DataType, 4>::const_pointer, size_t)’
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h:140:33: note: candidates are:
In file included from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h:101:0,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/strings/str_util.h:23,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:29,
                 from hungarian.cc:2:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:244:12: note: template<class C> tensorflow::gtl::array_slice_internal::ArraySliceImpl::ArraySliceImpl(const C&)
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:244:12: note:   template argument deduction/substitution failed:
In file included from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/strings/str_util.h:23:0,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:29,
                 from hungarian.cc:2:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h:140:33: note:   candidate expects 1 argument, 2 provided
In file included from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h:101:0,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/strings/str_util.h:23,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:29,
                 from hungarian.cc:2:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:230:7: note: constexpr tensorflow::gtl::array_slice_internal::ArraySliceImpl<tensorflow::DataType>::ArraySliceImpl(const tensorflow::gtl::array_slice_internal::ArraySliceImpl<tensorflow::DataType>&)
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:230:7: note:   candidate expects 1 argument, 2 provided
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:230:7: note: constexpr tensorflow::gtl::array_slice_internal::ArraySliceImpl<tensorflow::DataType>::ArraySliceImpl(tensorflow::gtl::array_slice_internal::ArraySliceImpl<tensorflow::DataType>&&)
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:230:7: note:   candidate expects 1 argument, 2 provided
In file included from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/strings/str_util.h:23:0,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:29,
                 from hungarian.cc:2:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h: In instantiation of ‘tensorflow::gtl::ArraySlice<T>::ArraySlice(std::initializer_list<_Tp>) [with T = long long int]’:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/tensor_shape.h:53:53:   required from here
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h:162:34: error: no matching function for call to ‘tensorflow::gtl::array_slice_internal::ArraySliceImpl<long long int>::ArraySliceImpl(std::initializer_list<long long int>::const_iterator, std::initializer_list<long long int>::size_type)’
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h:162:34: note: candidates are:
In file included from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h:101:0,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/strings/str_util.h:23,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:29,
                 from hungarian.cc:2:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:244:12: note: template<class C> tensorflow::gtl::array_slice_internal::ArraySliceImpl::ArraySliceImpl(const C&)
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:244:12: note:   template argument deduction/substitution failed:
In file included from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/strings/str_util.h:23:0,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:29,
                 from hungarian.cc:2:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h:162:34: note:   candidate expects 1 argument, 2 provided
In file included from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h:101:0,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/strings/str_util.h:23,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:29,
                 from hungarian.cc:2:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:230:7: note: constexpr tensorflow::gtl::array_slice_internal::ArraySliceImpl<long long int>::ArraySliceImpl(const tensorflow::gtl::array_slice_internal::ArraySliceImpl<long long int>&)
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:230:7: note:   candidate expects 1 argument, 2 provided
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:230:7: note: constexpr tensorflow::gtl::array_slice_internal::ArraySliceImpl<long long int>::ArraySliceImpl(tensorflow::gtl::array_slice_internal::ArraySliceImpl<long long int>&&)
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:230:7: note:   candidate expects 1 argument, 2 provided
In file included from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/strings/str_util.h:23:0,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:29,
                 from hungarian.cc:2:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h: In instantiation of ‘tensorflow::gtl::ArraySlice<T>::ArraySlice(const tensorflow::gtl::InlinedVector<T, N>&) [with int N = 4; T = long long int]’:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/partial_tensor_shape.h:101:53:   required from here
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h:140:33: error: no matching function for call to ‘tensorflow::gtl::array_slice_internal::ArraySliceImpl<long long int>::ArraySliceImpl(tensorflow::gtl::InlinedVector<long long int, 4>::const_pointer, size_t)’
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h:140:33: note: candidates are:
In file included from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h:101:0,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/strings/str_util.h:23,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:29,
                 from hungarian.cc:2:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:244:12: note: template<class C> tensorflow::gtl::array_slice_internal::ArraySliceImpl::ArraySliceImpl(const C&)
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:244:12: note:   template argument deduction/substitution failed:
In file included from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/strings/str_util.h:23:0,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:29,
                 from hungarian.cc:2:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h:140:33: note:   candidate expects 1 argument, 2 provided
In file included from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h:101:0,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/strings/str_util.h:23,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:29,
                 from hungarian.cc:2:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:230:7: note: constexpr tensorflow::gtl::array_slice_internal::ArraySliceImpl<long long int>::ArraySliceImpl(const tensorflow::gtl::array_slice_internal::ArraySliceImpl<long long int>&)
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:230:7: note:   candidate expects 1 argument, 2 provided
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:230:7: note: constexpr tensorflow::gtl::array_slice_internal::ArraySliceImpl<long long int>::ArraySliceImpl(tensorflow::gtl::array_slice_internal::ArraySliceImpl<long long int>&&)
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:230:7: note:   candidate expects 1 argument, 2 provided
In file included from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/strings/str_util.h:23:0,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:29,
                 from hungarian.cc:2:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h: In instantiation of ‘tensorflow::gtl::ArraySlice<T>::ArraySlice() [with T = std::pair<std::basic_string<char>, tensorflow::FunctionDefHelper::AttrValueWrapper>]’:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/function.h:95:32:   required from here
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h:127:34: error: no matching function for call to ‘tensorflow::gtl::array_slice_internal::ArraySliceImpl<std::pair<std::basic_string<char>, tensorflow::FunctionDefHelper::AttrValueWrapper> >::ArraySliceImpl(std::nullptr_t, int)’
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h:127:34: note: candidates are:
In file included from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h:101:0,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/strings/str_util.h:23,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:29,
                 from hungarian.cc:2:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:244:12: note: template<class C> tensorflow::gtl::array_slice_internal::ArraySliceImpl::ArraySliceImpl(const C&)
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:244:12: note:   template argument deduction/substitution failed:
In file included from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/strings/str_util.h:23:0,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:29,
                 from hungarian.cc:2:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h:127:34: note:   candidate expects 1 argument, 2 provided
In file included from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice.h:101:0,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/strings/str_util.h:23,
                 from /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:29,
                 from hungarian.cc:2:
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:230:7: note: constexpr tensorflow::gtl::array_slice_internal::ArraySliceImpl<std::pair<std::basic_string<char>, tensorflow::FunctionDefHelper::AttrValueWrapper> >::ArraySliceImpl(const tensorflow::gtl::array_slice_internal::ArraySliceImpl<std::pair<std::basic_string<char>, tensorflow::FunctionDefHelper::AttrValueWrapper> >&)
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:230:7: note:   candidate expects 1 argument, 2 provided
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:230:7: note: constexpr tensorflow::gtl::array_slice_internal::ArraySliceImpl<std::pair<std::basic_string<char>, tensorflow::FunctionDefHelper::AttrValueWrapper> >::ArraySliceImpl(tensorflow::gtl::array_slice_internal::ArraySliceImpl<std::pair<std::basic_string<char>, tensorflow::FunctionDefHelper::AttrValueWrapper> >&&)
/usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/lib/gtl/array_slice_internal.h:230:7: note:   candidate expects 1 argument, 2 provided
make: *** [hungarian/hungarian.so] Error 1
```

**Comment by Zehaos** [comment]
@Russell91 , I have fixed this error by using g++ 4.8 instead.
